lib/font/embedded.coffee

```diff
@@ -72,18 +72,18 @@ class EmbeddedFont extends PDFFont
       res.push ('0000' + gid.toString(16)).slice(-4)
 
       @widths[gid] ?= glyph.advanceWidth * @scale
-      @unicode[gid] ?= glyph.codePoints
+      @unicode[gid] ?= @font._cmapProcessor.codePointsForGlyph(glyph.id)
 
     return [res, positions]
 
-  encode2: (glyphs) ->
+  encodeGlyphs: (glyphs) ->
     res = []
     for glyph, i in glyphs
       gid = @subset.includeGlyph glyph.id
       res.push ('0000' + gid.toString(16)).slice(-4)
 
       @widths[gid] ?= glyph.advanceWidth * @scale
-      # @unicode[gid] ?= glyph.codePoints
+      @unicode[gid] ?= @font._cmapProcessor.codePointsForGlyph(glyph.id)
```

Comment:
The problem with this is that sometimes a glyph might represent more than one character. Also, multiple character sequences may map to the same glyph. I think we are going to need to move away from a `ToUnicode` map and try using an `ActualText` entry instead, which allows just putting the unicode string in the document alongside the text showing operator. This is documented in section 10.8.3 of the PDF 1.7 spec.

This may not compress as well, however, since the unicode mapping is duplicated every time the same glyph is used. Perhaps as an optimization later we could detect when a glyph is used twice in a different way and only generate `ActualText` in that case. However, for now we can just always use `ActualText` IMO.

Reply:
Note that I haven't experimented with `ActualText` at all, so I'm not sure how well supported it is across PDF readers, or how well it works. If you'd like to play around with it that would be cool